src/main/java/org/sonatype/nexus/ci/iq/ManualApplication.groovy

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.iq
+
+import hudson.Extension
+import org.jenkinsci.Symbol
+import org.kohsuke.stapler.DataBoundConstructor
+
+class ManualApplication
```

Comment:
For my own knowledge, I presume that the need for the `IqApplication`/`ManualApplication`/`SelectedApplication` class hierarchy is because of the need to play nicely with the Jenkins UI?

Reply:
> For my own knowledge, I presume that the need for the IqApplication/ManualApplication/SelectedApplication class hierarchy is because of the need to play nicely with the Jenkins UI?

Yes, SelectApplication and ManualApplication are because of the GUI.  You can create configs without them.  String is still supported.  But if you use the GUI they will be used and generated in the configs.

Reply:
The gui configs for each class are here:

https://github.com/sonatype/nexus-platform-plugin/blob/CLM-9736_Jenkins-Auto-App-Creation/src/main/resources/org/sonatype/nexus/ci/iq/ManualApplication/config.groovy

https://github.com/sonatype/nexus-platform-plugin/blob/CLM-9736_Jenkins-Auto-App-Creation/src/main/resources/org/sonatype/nexus/ci/iq/SelectedApplication/config.groovy